### PR TITLE
🐛 Destination Snowflake - change typing and deduping error column creation

### PR DESCRIPTION
**Pull request by jbfbell**
Attempting to reduce the complexity of our typing and deduping queries, specifically removing chained `ARRAY_CAT` calls in favor of `ARRAY_CONSTRUCT_COMPACT`. On a small scale (500 rows) the total duration of the query dropped from 938ms to 743ms (~20% faster) but maybe more interesting/relevant is that the SQL compilation time dropped from 476ms (50% of 938ms) to 202ms (27% of 743ms) or roughly 57% faster.

### CODE REVIEW

File: docs/integrations/destinations/snowflake.md [hunk 1]
```diff
@@ -271,6 +271,7 @@ Otherwise, make sure to grant the role the required permissions in the desired n
 
 | Version         | Date       | Pull Request                                               | Subject                                                                                                                                                         |
 |:----------------|:-----------|:-----------------------------------------------------------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 2.1.5           | 2023-08-28 | [\#29903](https://github.com/airbytehq/airbyte/pull/29917) | Destinations V2: Change Metadata error array construction from ARRAY_CAT to ARRAY_CONSTRUCT_COMPACT                                                             |
```
Comment (edgao) [inline docs/integrations/destinations/snowflake.md:274]: Nit: maybe mention that this should be a performance improvement?

**Review by edgao — APPROVED** [review]
:shipit: pending CI success

**Review by evantahler — APPROVED** [review]
Seems legit!

Screenshot 2023-08-28 at 5 37 50 PM

**Comment by jbfbell** [comment]
Relates to https://github.com/airbytehq/oncall/issues/2820

**Pull request merged**